tests/integration_tests/performance/test_memory_overhead.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Track Firecracker memory overhead
+
+Because Firecracker is a static binary, and is copied before execution, no
+memory is shared across many different processes. It is thus important to track
+how much memory overhead Firecracker adds.
+
+These tests output metrics to capture the memory overhead that Firecracker adds,
+both from the binary file (file-backed) and what Firecracker allocates during
+the process lifetime.
+
+The memory overhead of the jailer is not important as it is short-lived.
+"""
+
+from collections import defaultdict
+from pathlib import Path
+
+import psutil
+import pytest
+
+from framework.properties import global_props
+
+# If guest memory is >3328MB, it is split in a 2nd region
+X86_MEMORY_GAP_START = 3328 * 2**20
+
+
+@pytest.mark.parametrize(
+    "vcpu_count,mem_size_mib",
+    [(1, 128), (1, 1024), (2, 2024), (4, 4096)],
+)
+def test_memory_overhead(
+    microvm_factory, guest_kernel, rootfs, vcpu_count, mem_size_mib, metrics
+):
+    """Track Firecracker memory overhead.
+
+    We take a single measurement as it only varies by a few KiB each run.
+    """
+
+    microvm = microvm_factory.build(guest_kernel, rootfs)
+    microvm.spawn()
+    microvm.basic_config(vcpu_count=vcpu_count, mem_size_mib=mem_size_mib)
+    microvm.add_net_iface()
+    microvm.start()
+
+    # check that the vm is running
+    microvm.ssh.run("true")
+
+    guest_mem_bytes = mem_size_mib * 2**20
+    guest_mem_splits = {
+        guest_mem_bytes,
+        X86_MEMORY_GAP_START,
+        guest_mem_bytes - X86_MEMORY_GAP_START,
+    }
+    mem_stats = defaultdict(int)
+    mem_stats["guest_memory"] = guest_mem_bytes
+    ps = psutil.Process(microvm.jailer_clone_pid)
```

Comment:
So, this gathers statics for microVMs, that run inside the jailer. Maybe we should put that in the comment on the top of this file.

Reply:
`jailer_clone_pid` is the Firecracker PID. Since the jailer exec()s into Firecracker, the jailer doesn't play a role, I think?

Reply:
Would it make sense to have 2 tests?

1. Firecracker
2. Firecracker + Jailer

I think both metrics may be useful.

Reply:
I don't see how the jailer would play a role?

Reply:
> I don't see how the jailer would play a role?

Could we then run the test without the jailer?

I think if we aren't concerned with testing it here maybe we can remove this complexity.

Reply:
Well, two reasons:
1. it is the default
2. the jailer it's the bit that makes sure different Firecracker processes don't share pages, buy copying the binary before executing it

Reply:
I think in the case where the jailer does affect the memory size of Firecracker even as a result of side affects it makes sense to measure this.

Although I am okay to merge this without this.